cpp/src/arrow/compute/exec/aggregate.cc

```diff
@@ -38,33 +38,39 @@ namespace internal {
 
 Result<std::vector<const HashAggregateKernel*>> GetKernels(
     ExecContext* ctx, const std::vector<Aggregate>& aggregates,
-    const std::vector<TypeHolder>& in_types) {
+    const std::vector<std::vector<TypeHolder>>& in_types) {
   if (aggregates.size() != in_types.size()) {
     return Status::Invalid(aggregates.size(), " aggregate functions were specified but ",
                            in_types.size(), " arguments were provided.");
   }
 
   std::vector<const HashAggregateKernel*> kernels(in_types.size());
 
+  std::vector<TypeHolder> aggregate_in_types;
   for (size_t i = 0; i < aggregates.size(); ++i) {
     ARROW_ASSIGN_OR_RAISE(auto function,
                           ctx->func_registry()->GetFunction(aggregates[i].function));
+    // {in_types[i]..., uint32()}
+    aggregate_in_types.reserve(in_types[i].size() + 1);
```

Comment:
```suggestion
```
I'm not sure what purpose this `reserve` has

Reply:
vector copy assignment should do a reserve internally and if it doesn't, they probably have a good reason not to :)

Reply:
nvm, I see.  You are copying and adding one.  Still, I think it's simpler without the reserve.